### PR TITLE
Bug 1943845: Add startup probe to the router deployment

### PR DESCRIPTION
**Pull request by Miciah**
Add a startup probe to allow the router up to 2 minutes start up before sending liveness and readiness probes in case initial synchronization takes a substantial amount of time.  Using a startup probe instead of specifying `initialDelaySeconds` on the liveness and readiness probes allows the kubelet to start performing liveness and readiness probes quickly if the initial synchronization is quick.

* `assets/router/deployment.yaml`: Add startup probe.  Delete `initialDelaySeconds` for the liveness and readiness probes.
* `pkg/manifests/bindata.go`: Regenerate.
* `pkg/operator/controller/ingress/deployment.go` (`desiredRouterDeployment`): Configure the startup probe.
(`hashableDeployment`): Hash the startup probe.
(`deploymentConfigChanged`): Update the startup probe.
* `pkg/operator/controller/ingress/deployment_test.go` (`TestDesiredRouterDeployment`): Verify that the startup probe is configured as expected.  Fix an incorrect error message for a readiness probe check.
(`TestDeploymentConfigChanged`): Split the liveness and readiness test cases to verify that an update either one alone is detected as a change.  Add a test case for the startup probe.

### CODE REVIEW

**Comment by Miciah** [comment]
`TestUniqueIdHeader` failed.  
/test e2e-aws-operator

**Comment by Miciah** [comment]
SDN/OVS didn't initialize on all nodes.
/test e2e-aws-operator

File: pkg/operator/controller/ingress/deployment_test.go [hunk 1]
```diff
@@ -928,6 +951,17 @@ func TestDeploymentConfigChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if probe values are set to non-default values",
```
Comment (sgreene570) [inline pkg/operator/controller/ingress/deployment_test.go:955]:
```suggestion
			description: "if startup probe values are set to non-default values",
```

Comment (Miciah) [reply pkg/operator/controller/ingress/deployment_test.go]: Updated.  Thanks!

**Comment by sgreene570** [comment]
Looks great!

/lgtm

**Comment by Miciah** [comment]
The e2e-aws- job failed because the "openshift-kube-controller-manager/installer-7-ip-10-0-169-162.us-west-2.compute.internal" pod was crashlooping, logging `dial tcp 172.30.0.1:443: connect: connection refused`.

/test e2e-aws

The e2e-upgrade job failed because the kubelet failed to create a sandbox for the "loki/loki-promtail-pjbq7" pod because Multus failed to connect to the API, logging "dial tcp 10.0.246.156:6443: connect: connection refused".

/test e2e-upgrade

**Comment by Miciah** [comment]
"Failed to create pod sandbox" for pods "openshift-multus/network-metrics-daemon-wj6ff" and "openshift-dns/dns-default-gpmtr" because Multus failed with "dial tcp 10.0.226.53:6443: connect: connection refused".

/test e2e-upgrade

**Pull request merged**